miqcli/constants.py

```diff
@@ -25,8 +25,15 @@
 #: name of miqcli package
 PACKAGE = 'miqcli'
 
-#: Installed version of miqcli
```

Comment:
I am still unsure about using pbr `get_version` function within the constants module to set the version. Each time the CLI is run it will use pbr to get the version information. Pbr's function tries 3 different locations to get this information.

1. By an environment variable
2. From the package metadata (distribution)
3. From git

I created a test package on the PyPI test server with a very simple Python package setup using pbr. Once I installed the package and tried to run. I was unable to proceed, I was getting the same exception mentioned in #60. My opinion is when this is live and ready on PyPI, we will not be able to get the version from pbr. It is going to have to be defined within the constants module.

Reply:
The version should never be statically defined in the constants module, lest we be forced to update it constantly. One of the biggest reasons to use pbr is that your version is abstracted and you never again have to care about it except for the purposes of tagging releases in git. My usage in this case is just an abuse of pbr, because (as the name would imply despite my ignorance) pbr is for packaging, not...not packaging(?).

The version should always be detectable from the environment, and this appears to work reliably when going through `pkg_resources` to do it. I just gave up too easily when it didn't work once for me, and I never noticed that I was using *the wrong python interpreter*.

Reply:
I should have mentioned in my comment that this is first time working with pbr. You are right, having the capability provided by pbr to handle the versioning is great!

Reply:
I've been using pbr pretty happily for years, and frankly should have known better when I encountered that error with the "setuptoolsy" `get_distribution` rather than go searching for some other way to do it. I'm very glad you caught this quickly.